circleci.sh

```diff
@@ -97,9 +98,17 @@ coverage()
     make -f posix.mak $(find std etc -name "*.d" | sed "s/[.]d$/.test/" | grep -vE '(std.algorithm.sorting|std.encoding|net.curl)' )
 }
 
+# compile all public unittests separately
+publictests()
+{
+    clone https://github.com/dlang/tools.git ../tools master
```

Comment:
Why master instead of $base_branch? I'd argue that unconditionally using master is almost always wrong, b/c it's not tied to the phobos version.
In fact we were only able to resolve most of our testing issues, after scrutinizingly pinning all our dmd, dub, et.al. versions.
Master also means that repeating builds might not be reproducible, that's just asking for troubles (e.g. I spend about 1 week on [Issue 16574](https://issues.dlang.org/show_bug.cgi?id=16574) b/c the reporter only provided a moving target).

Possible ways to avoid future issues:
- checkout a specific version (v2.072.2) or commit of that file in the tools repo
- maintain that tool in phobos if it's merely designed for phobos
- do dmd independent releases of that script and pin those